### PR TITLE
declare UCX and HCOLL as externals for Open MPI installation using Spack

### PR DESCRIPTION
**Pull request by abhamidipati0614**
- Reference : https://spack.readthedocs.io/en/latest/build_settings.html#prevent-packages-from-being-built-from-sources

### CODE REVIEW

File: ubuntu/common/install_mpis.sh [hunk 1]
```diff
@@ -20,14 +20,89 @@ tar -xvf $tarball
...
 
+# Module Files
+module_files_directory=/usr/share/modules/modulefiles/mpi
+mkdir -p $module_files_directory
+
+# HPC-X
+cat << EOF >> $module_files_directory/hpcx-$hpcx_version
+#%Module 1.0
+#
+#  HPCx $hpcx_version
```
Comment (moes1) [inline ubuntu/common/install_mpis.sh:31]: What's going on with the indenting here?

**Review by moes1 — APPROVED** [review]

File: ubuntu/common/install_nvidiagpudriver.sh [hunk 2]
```diff
@@ -24,7 +24,7 @@ $COMMON_DIR/download_and_verify.sh $cuda_samples_download_url $cuda_samples_sha2
 tar -xvf $tarball
 pushd ./cuda-samples-$cuda_samples_version
 make
-mv -vT ./Samples /usr/local/cuda-${CUDA_SAMPLES_VERSION}/samples
+mv -vT ./Samples /usr/local/cuda/samples
```
Comment (rafsalas19) [inline ubuntu/common/install_nvidiagpudriver.sh:27]: Do we have anything dependent on this old path?

File: ubuntu/common/install_mpis.sh (hunk 1 above)
Comment (abhamidipati0614) [reply ubuntu/common/install_mpis.sh:31]: This is a module file specification. Please refer - https://modules.readthedocs.io/en/stable/modulefile.html

File: ubuntu/common/install_nvidiagpudriver.sh (hunk 2 above)
Comment (abhamidipati0614) [reply ubuntu/common/install_nvidiagpudriver.sh:27]: No we don't

File: ubuntu/common/install_mpis.sh [hunk 3]
```diff
@@ -105,8 +166,7 @@ setenv          MPI_MAN         $(echo $impi_2021_path)/man
 setenv          MPI_HOME        $(echo $impi_2021_path)
 EOF
 
-# Softlinks
-ln -s $module_files_directory/hpcx-$hpcx_version $module_files_directory/hpcx
```
[the remark this answers is not in the document]
Comment (abhamidipati0614) [reply ubuntu/common/install_mpis.sh:109]: It is established before this line (right after HPC-x installation to gather the UCX and HCOLL paths)

**Pull request merged**